File: guides/micronaut-error-handling/java/src/main/java/example/micronaut/CommandBookSave.java
package example.micronaut;

import io.micronaut.serde.annotation.Serdeable;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;

@Serdeable // <1>
public class CommandBookSave {

    @NotBlank // <2>
    private String title;

    @Positive // <3>
    private int pages;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getPages() {
        return pages;
    }

    public void setPages(int pages) {
        this.pages = pages;
    }
}

File: guides/micronaut-error-handling/kotlin/src/main/kotlin/example/micronaut/CommandBookSave.kt
package example.micronaut

import io.micronaut.serde.annotation.Serdeable
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Positive

@Serdeable // <1>
data class CommandBookSave(
    @field:NotBlank val title: String, // <2>
    @field:Positive val pages: Int // <3>
)